### PR TITLE
dash/hccColumnAssignment-nulls

### PR DESCRIPTION
**Pull request by sebastianbochan**
Replaced blacklist with whitelist in columnAssignment option.

--- 

- [x] - fix
- [x] - update demos

### CODE REVIEW

File: samples/dashboards/demo/climate/demo.js [hunk 1]
```diff
@@ -1064,13 +1065,13 @@ async function updateBoard(board, city, column, scale, newData) {
...
-                TNF: column === 'TNF' ? 'y' : void 0,
-                TXC: column === 'TXC' ? 'y' : void 0,
-                TXF: column === 'TXF' ? 'y' : void 0
+                FD: column === 'FD' && 'y',
+                ID: column === 'ID' && 'y',
+                RR1: column === 'RR1' && 'y',
+                TNC: column === 'TNC' && 'y',
+                TNF: column === 'TNF' && 'y',
+                TXC: column === 'TXC' && 'y',
+                TXF: column === 'TXF' && 'y'
```
Comment (bre1470) [inline samples/dashboards/demo/climate/demo.js:1074]: This is not type safe. You should settle on something regarding the negative flag. At the moment it is `null`, previously it was `undefined`, here it is `false`.

File: samples/dashboards/demo/climate/demo.js [hunk 2]
```diff
@@ -1083,13 +1084,13 @@ async function updateBoard(board, city, column, scale, newData) {
...
-            TNF: column === 'TNF' ? 'y' : void 0,
-            TXC: column === 'TXC' ? 'y' : void 0,
-            TXF: column === 'TXF' ? 'y' : void 0
+            FD: column === 'FD' && 'y',
+            ID: column === 'ID' && 'y',
+            RR1: column === 'RR1' && 'y',
+            TNC: column === 'TNC' && 'y',
+            TNF: column === 'TNF' && 'y',
+            TXC: column === 'TXC' && 'y',
+            TXF: column === 'TXF' && 'y'
```
Comment (bre1470) [inline samples/dashboards/demo/climate/demo.js:1093]: Same issue here.

File: ts/Dashboards/Plugins/HighchartsComponent.ts [hunk 3]
```diff
@@ -283,7 +284,7 @@ class HighchartsComponent extends Component {
                         'chartConfig'
                     ]
                 }),
-            columnAssignment: {}
+            columnAssignment: null
```
Comment (bre1470) [inline ts/Dashboards/Plugins/HighchartsComponent.ts:287]: This is not type safe as well. Use `void 0` or remove it if possible.

File: ts/Dashboards/Plugins/HighchartsComponent.ts [hunk 4]
```diff
@@ -618,11 +619,11 @@ class HighchartsComponent extends Component {
...
                         true;
 
-                    if (!isVisible && !columnAssignment[name]) {
-                        return false;
+                    if (!defined(this.options.columnAssignment)) {
+                        return true;
                     }
 
-                    if (columnAssignment[name] === null) {
+                    if (!isVisible || !columnAssignment[name]) {
```
Comment (bre1470) [inline ts/Dashboards/Plugins/HighchartsComponent.ts:626]: This will have issues with empty strings.

**Review by bre1470 — CHANGES_REQUESTED** [review]

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Please update the API for `columnAssignment`

File: samples/dashboards/demo/climate/demo.js (hunk 1 above)
Comment (sebastianbochan) [reply samples/dashboards/demo/climate/demo.js:1074]: You're right, I optimized it, too much. Let's leave nulls there.

File: samples/dashboards/demo/climate/demo.js (hunk 2 above)
Comment (sebastianbochan) [reply samples/dashboards/demo/climate/demo.js:1093]: You're right, I optimized it, too much. Let's leave nulls there.

File: ts/Dashboards/Plugins/HighchartsComponent.ts (hunk 4 above)
Comment (bre1470) [reply ts/Dashboards/Plugins/HighchartsComponent.ts:626]: I will create a ticket.

File: ts/Dashboards/Plugins/HighchartsComponent.ts [hunk 5]
```diff
@@ -993,8 +994,8 @@ namespace HighchartsComponent {
          */
         chartID?: string;
         /**
-         * Names / aliases that should be mapped to xAxis values. You can use
-         * null to keep columns selectively out of the chart.
+         * Names / aliases that should be mapped to xAxis values.
```
Comment (karolkolodziej) [inline ts/Dashboards/Plugins/HighchartsComponent.ts:997]: Sorry for being picky but I would describe how it works. What is going to happen if you don't declare them etc.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

File: ts/Dashboards/Plugins/HighchartsComponent.ts (hunk 4 above)
Comment (sebastianbochan) [reply ts/Dashboards/Plugins/HighchartsComponent.ts:626]: I tested this scenario and all should work properly. If you have data or scenario when not, please let me know.

**Review by pawellysy — APPROVED** [review]
LGTM

**Review by karolkolodziej — APPROVED** [review]

**Pull request merged**